src/proper_target.erl

```diff
@@ -265,20 +303,35 @@ init([{Strategy, Data}]) ->
 handle_call(gen, _From, State) ->
   #state{strategy = Strat, target = Target, data = Data} = State,
   {NextValue, NewTarget, NewData} = Strat:next(Target, Data),
-  {reply, NextValue, State#state{target = NewTarget, data = NewData}};
+  Ret = case State#state.stateful of
```

Comment:
Why is this not part of the generator? What makes stateful so different that you need this extra code-path. For non-targeted generators this is not the case and having it for targeted ones is a bit strange.

Reply:
For stateful targeted property testing, what two values are generated, one containing a `map` with command names as keys and weight as a value, and the other containing the actual commands. The actual search occurs on the weights of each command, instead of directly on the commands generated. However, I did not want to alter the already existing API of stateful commands, that is why internally I need to keep both values, while the API needs only return the command sequence to `proper` and subsequently the user.